### PR TITLE
Add setSupportedTokens

### PR DESCRIPTION
**Pull request by cehali**
# Summary
Add setSupportedTokens, move DeploymentReadyObserver to FutureWalletFactory

### CODE REVIEW

File: universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx [hunk 1]
```diff
@@ -29,20 +29,21 @@ export interface ChooseTopUpMethodProps {
...
-  }
-  const gasParametersChanged = (gasParameters: GasParameters) => {
-    setGasParameters(gasParameters);
-    onGasParametersChanged!(gasParameters);
+export const ChooseTopUpMethod = ({sdk, contractAddress, onPayClick, topUpClassName, logoColor, isDeployment, walletService}: ChooseTopUpMethodProps) => {
+  const [minimalAmount, setMinimalAmount] = useState('0');
+
+  const onGasParametersChanged = (gasParameters: GasParameters) => {
+    ensureNotNull(walletService, MissingParameter, 'walletService');
+    ensure(walletService.state.kind === 'Future', InvalidWalletState, 'Future', walletService.state.kind);
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx]: Move to walletService.setGasParameters

File: universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx [hunk 2]
```diff
@@ -36,9 +36,10 @@ export const ChooseTopUpMethod = ({sdk, contractAddress, onPayClick, topUpClassN
 
   const onGasParametersChanged = (gasParameters: GasParameters) => {
     ensureNotNull(walletService, MissingParameter, 'walletService');
-    setMinimalAmount(safeMultiply(MINIMAL_DEPLOYMENT_GAS_LIMIT, gasParameters.gasPrice));
+    const minimalAmountCalculated = safeMultiply(MINIMAL_DEPLOYMENT_GAS_LIMIT, gasParameters.gasPrice);
+    setMinimalAmount(minimalAmountCalculated);
     walletService.setGasParameters(gasParameters);
-    walletService.setSupportedTokens([{address: gasParameters.gasToken, minimalAmount}]);
+    walletService.setSupportedTokens([{address: gasParameters.gasToken, minimalAmount: minimalAmountCalculated}]);
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx]:
```ts
const onGasParametersChanged = (gasParameters: GasParameters) => {
  walletService.setGasParameters(gasParameters);
  setMinimalAmount(walletService.getRequiredDeploymentBalance());
}
```

Comment (marekkirejczyk) [reply universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx]: inline `setSupportedTokens` to `setGasParameters`

File: universal-login-sdk/lib/core/observers/DeploymentReadyObserver.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export class DeploymentReadyObserver extends ObserverRunner {
     if (tokensWithValidMinimaAmount.length > 0) {
       this.supportedTokens = supportedTokens;
     }
+    return tokensWithValidMinimaAmount;
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/observers/DeploymentReadyObserver.ts]: Unnecessary

File: universal-login-sdk/test/integration/core/DeploymentReadyObserver.ts [hunk 4]
```diff
@@ -59,6 +59,18 @@ describe('INT: DeploymentReadyObserver', () => {
       .to.be.rejectedWith('Other wallet waiting for counterfactual deployment. Stop observer to cancel old wallet instantialisation.');
   });
 
+  it('#setSupportedTokens, minimalAmount > 9', () => {
+    expect(deploymentReadyObserver.setSupportedTokens(supportedTokens)).to.deep.eq(supportedTokens);
+  });
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/integration/core/DeploymentReadyObserver.ts]: Unecessary

File: universal-login-react/src/ui/Onboarding/OnboardingSteps.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ export function OnboardingSteps({sdk, walletService, className, onCreate}: Onboa
         <TopUp
           modalClassName={className}
           contractAddress={walletState.wallet.contractAddress}
-          onGasParametersChanged={(gasParameters) => walletService.setGasParameters(gasParameters)}
+          walletService={walletService}
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Onboarding/OnboardingSteps.tsx]: Keep `walletService.setGasParameters(gasParameters)`

File: universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx (hunk 1 above)
Comment (marekkirejczyk) [reply universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx]: ensure not needed

File: universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx [hunk 6]
```diff
@@ -29,20 +24,18 @@ export interface ChooseTopUpMethodProps {
   topUpClassName?: string;
   logoColor?: LogoColor;
   isDeployment: boolean;
-  onGasParametersChanged?: OnGasParametersChanged;
+  walletService?: WalletService;
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx]: we can make it required

**Pull request merged**